### ISSUE TITLE
Separate model fitting step from get_regresion_x() functions

### ISSUE THREAD

**Issue by rudeboybert** [moderndive/moderndive#4]
```
library(moderndive)
library(dplyr)
mtcars <- mtcars %>% 
  mutate(cyl = as.factor(cyl))

# instead of doing this
get_regression_table(mpg ~ hp, data = mtcars, digits = 3, print = FALSE)
get_regression_points(mpg ~ hp + cyl, data = mtcars)
get_regression_summaries(mpg ~ hp, data = mtcars)

# do this
mpg_model <- lm(mpg ~ hp, data = mtcars)
get_regression_table(mpg_model, digits = 3, print = FALSE)
get_regression_points(mpg_model)
get_regression_summaries(mpg_model)
```

**Comment by ismayc** [comment]
Got to 100% test coverage and implemented this at https://github.com/moderndive/moderndive/pull/5

**Comment by rudeboybert** [comment]
Resolved by #5
